### PR TITLE
Add simple Delete folder btn

### PR DESCRIPTION
**Pull request by vidma**
resolves https://github.com/spark-notebook/spark-notebook/issues/928

as this does a recursive delete, it would be nice to add more permissions here so people could delete only what belongs to them...

### CODE REVIEW

File: public/ipython/tree/js/notebooklist.js [hunk 1]
```diff
@@ -602,6 +605,42 @@ define([
         item.find(item_buttons_selector).append(delete_button);
     };
 
+  NotebookList.prototype.add_delete_directory_button = function (item) {
```
Comment (vidma) [inline public/ipython/tree/js/notebooklist.js]: same as `add_delete_btn` below

**Comment by vidma** [comment]
cc @antonkulaga

**Comment by vidma** [comment]
ready to 🚢 .

P.S. deleting folders is rather dangerous, maybe we need a config option for it...

**Comment by frbayart** [comment]
@vidma did you seen the failed checks ?

```[info] complete

<console>:16: error: value toS is not a member of snippet.AnExample

test.toS

     ^

<console>:16: error: value toS is not a member of snippet.AnExample

test.toS

     ^

[info] - should return the correct completions

<console>:16: error: value testMeth is not a member of snippet.AnExample

test.testMeth

     ^

[info] - should lists all overrided method versions, indicating optional parameters if any

<console>:17: error: identifier expected but '}' found.

}

^

Set("==(x$1: Any)", "+(other: String)", "testMethod(a: String, b: String)", "finalize", "notify", "equals(x$1: Any)", "<init>(aField: Int, [notAField: Boolean])", "!=(x$1: Any)", "implicitMethod(a: Int)", "clone", "ensuring(cond: snippet.AnExample => Boolean)", "hashCode", "asInstanceOf", "testVar", "wait(x$1: Long, x$2: Int)", "aField", "##", "formatted(fmtstr: String)", "->(y: B)", "cls", "toSchwarz", "→(y: B)", "<init>(cls: snippet.AnExample)", "ensuring(cond: Boolean)", "wait", "testMethod(a: Int, [optionalB: String])", "ne(x$1: AnyRef)", "ensuring(cond: snippet.AnExample => Boolean, msg: => Any)", "synchronized(x$1: T0)", "getClass", "notAField", "eq(x$1: AnyRef)", "testMethod(a: String)", "toString", "ensuring(cond: Boolean, msg: => Any)", "isInstanceOf", "wait(x$1: Long)", "notifyAll")

```

**Pull request merged**